android/gltfio-android/src/main/cpp/MaterialKey.cpp

```diff
@@ -72,6 +72,32 @@ Java_com_google_android_filament_gltfio_MaterialProvider_00024MaterialKey_nGloba
     MaterialKeyHelper::get().init(env);
 }
 
+extern "C"
+JNIEXPORT void JNICALL
+Java_com_google_android_filament_gltfio_MaterialProvider_00024MaterialKey_nConstrainMaterial(JNIEnv* env, jclass,
+        jobject materialKey, jintArray uvMap) {
+    MaterialKey nativeMaterialKey = {};
+
+    auto& helper = MaterialKeyHelper::get();
+    helper.copy(env, nativeMaterialKey, materialKey);
+
+    UvMap nativeUvMap = {};
+    constrainMaterial(&nativeMaterialKey, &nativeUvMap);
+
+    // Copy the UvMap results from the native array into the JVM array.
+    jint* elements = env->GetIntArrayElements(uvMap, nullptr);
+    if (elements) {
+        const size_t javaSize = env->GetArrayLength(uvMap);
```

Comment:
Use `std::copy_n` instead. For example:

```
std::copy_n(asset->getEntities(),
            std::min(available, (jsize) asset->getEntityCount()), entities);
```

Reply:
Sure, but I've copied that snippet from the other part of Filament, literally neighboring file...

Reply:
Ugh, copy_n won't work - types are different :|

Reply:
Argh, sorry about that. That explains the original code :(